### PR TITLE
[Hub Generated] Review request for Azure.Analytics.Purview.Catalog to add version stable/2023-09-01

### LINKED ISSUE #25412: [Microsoft Purview ] API Review
New API Review meeting has been requested.

**Service Name**: Microsoft Purview 
**Review Created By**: Yifan Zhou
**Review Date**: 08/31/2023 04:00 PM PT
**Onboarding Record**: 
**PR**: 
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=f21fa127-ee3f-ee11-bdf3-000d3a341d27)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by summerjune0730**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/summerjune0730-purview-Azure.Analytics.Purview.Catalog-2023-09-01...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [x] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Review by mikekistler — APPROVED** [review]
Looks good. 👍

**Comment by summerjune0730** [comment]
@MushMal , could you help to review this PR? Please let me know if any follow up needed from my side further. Thank you.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json [hunk 1]
```diff
@@ -0,0 +1,7267 @@
...
+        "operationId": "Entity_MoveEntitiesToCollection",
+        "x-ms-examples": {
+          "Collection_MoveEntitiesToCollection": {
+            "$ref": "./examples/Entity_MoveEntitiesToCollection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/requiredCollectionId"
+          },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:1524]: Where is the "collectionId" of the entity? Is it in path or in entity property?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json [hunk 2]
```diff
@@ -0,0 +1,7267 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/lineage/{guid}/next/": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:2994]: I am not sure about the purpose of this API. Is it a "List" or only the "{nextPage}" URL of another API?

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json [hunk 3]
```diff
@@ -0,0 +1,7267 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/types/termtemplatedef/guid/{guid}": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:3934]: I didn't see a PUT of `/types/termtemplatedef/guid/{guid}`. How's it created?

**Review by weidongxu-microsoft — COMMENTED** [review]
I've only reviewed the non-`/atlas/v2` APIs.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json (hunk 1 above)
Comment (summerjune0730) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:1524]: It's in the path. We only allow the customer to specify one collectionid for all input entities.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json (hunk 2 above)
Comment (summerjune0730) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:2994]: As the lineage info might contains too many entities, so this api is to get lineage with pagination support.

File: specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json (hunk 3 above)
Comment (summerjune0730) [reply specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json:3934]: all type defs are written by native Atlas API Post/Put /atlas/v2/types/typedefs

**Review by anuchandy — APPROVED** [review]

**Pull request merged**